src/utils/wallet.js

```diff
@@ -194,21 +203,78 @@ export async function getWalletFromStorage(storageData: Object, dispatch: Dispat
   };
 }
 
-export async function decryptWallet(encryptedWallet: Object, saltedPin: string) {
-  return ethers.Wallet.fromEncryptedJson(JSON.stringify(encryptedWallet), saltedPin);
-}
-
-export function constructWalletFromPrivateKey(privateKey: string): Object {
+export function constructWalletFromPrivateKey(privateKey: string): ethers.Wallet {
   return new ethers.Wallet(privateKey);
 }
 
-export function constructWalletFromMnemonic(mnemonic: string): Object {
+export function constructWalletFromMnemonic(mnemonic: string): ethers.Wallet {
   return ethers.Wallet.fromMnemonic(mnemonic);
 }
 
-export async function getPrivateKeyFromPin(pin: string, dispatch: Dispatch) {
+export async function decryptWalletFromStorage(pin: string, dispatch: Dispatch): Promise<ethers.Wallet> {
```

Comment:
Similar to comment for `getSaltedPin`: consider replacing props with `pin, deviceUniqueId` and do the DeviceInfo check at call site. Passing Dispatch to utils feels a bit mixing responsibilities.